guidelines/static-websites.md

```diff
@@ -0,0 +1,137 @@
+# Rust Infrastructure hosting for static websites
+
+The Rust Infrastructure team provides hosting for static websites available for
+all Rust teams. This document explains the requirements a website needs to meet
+and how to setup one.
+
+## Requirements for hosting websites
+
+* **The website must be managed by a Rust team, or be officially affiliated with
+  the project.**  
+  The infrastructure team has finite resources and we can't offer hosting for
+  community projects.
+* **The website’s content and build tooling must be hosted on a GitHub
+  repository in either the [rust-lang](https://github.com/rust-lang) or
+  [rust-lang-nursery](https://github.com/rust-lang-nursery) organizations.**  
+  The infrastructure team must be able to rebuild the website content at any
+  time (for example if we need to switch hosting), and having it hosted on a
+  GitHub repository inside infra-managed organizations is the best way for us
+  to ensure that. Even though we'd prefer for all the repositories to be public
+  it's not a requirement.
+* **The website must be built and deployed with Travis CI.**  
+  We have custom tooling built around hosting static websites on our infra,
+  and at the moment they only work with Travis CI. If you need different CI
+  services we can try to adapt our tools.
+* **The website must reach an A+ grade on the
+  [Mozilla Observatory](https://observatory.mozilla.org/).**  
+  Browsers have multiple security features toggleable only through HTTP
+  response headers, and those features enhance users' privacy and prevent
+  exploits from working. An A+ grade on the Observatory indicates all the
+  important headers are correctly set.
+* **The website can be hosted on either GitHub Pages or Amazon S3 (in the
+  rust-lang account), and must sit behind CloudFront.**  
```

Comment:
Yeah, that's true, but I'd also like to push to standardize how things are hosted as much as possible, with an "do as we say here unless you have a really good reason not to" attitude. Would a rewrite of this requirement with that in mind be ok for you?

Reply:
I think that makes sense yeah to strive for uniformity, but that largely just means that we're routing through cloudfront to some static file hosting service, which may be S3 or GitHub pages, but static file servers, if any, should also suffice.

(this is just a minor concern and doesn't matter too much)

Reply:
I'd actually drop this requirement in the sense that if it passes the observatory that's all we care about, and cloudfront is just a means to an end of getting an A+ grade.

Reply:
Changed the wording to "The website must be hosted on platforms vetted by the infra team" with GHPages/S3+CloudFront as our recommendation.